src/BatteryStateOfChargeService.cpp

```diff
@@ -43,21 +44,65 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY - totalAH_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    return isCharging_;
 }
 
+
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    QTime ntime(0,0,0,0);
+    if (time.isNull())
+        return ntime;
+    else
+        return time;
```

Comment:
You should consider moving some of the logic in `addData` into this function, `addData` is quite full right now and some of the logic done for this function is done in `addData`.